src/Downloads.jl

```diff
@@ -46,27 +75,33 @@ function download(
     downloader::Downloader = default_downloader(),
 )
     yield() # prevents deadlocks, shouldn't be necessary
-    arg_write(output) do io
-        easy = Easy()
-        set_url(easy, url)
-        for hdr in headers
-            hdr isa Pair{<:AbstractString, <:Union{AbstractString, Nothing}} ||
-                throw(ArgumentError("invalid header: $(repr(hdr))"))
-            add_header(easy, hdr)
-        end
-        add_handle(downloader.multi, easy)
-        for buf in easy.buffers
-            write(io, buf)
-        end
-        remove_handle(downloader.multi, easy)
-        status = get_response_code(easy)
-        status == 200 && return
-        if easy.code == Curl.CURLE_OK
-            message = get_response_headers(easy)[1]
-        else
-            message = GC.@preserve easy unsafe_string(pointer(easy.errbuf))
+    using_default = downloader === default_downloader()
```

Comment:
```suggestion
    using_default = downloader === DEFAULT_DOWNLOADER[]
```
This creates the default downloader on any download call, which seems like an unwanted side effect. We can just check if it's equal to `DEFAULT_DOWNLOADER[]`: if no default downloader has been created, this is `nothing` so the comparison is false; if a default downloader has been created, then this gets it.